### DISCUSSION TITLE
constraints: checking if a value belongs to a custom array?

### DISCUSSION THREAD

**Discussion by csyding** [flipt-io/flipt#3478] [Q&A]
hi there!

i was wondering if there's a good way to create a constraint where the type is an array and just check if contains a value? for example, in my use case, i might have a user that belongs to multiple groups:

"groups" : ["admin", "accounting", "underwriting"]

and for a flag, i might want anyone who has the group "accounting" assigned to them to evaluate to true. at the moment i've just been doing a loop over each of the groups and evaluating that way, but is there a better way to do this? i was hoping for something like "if groups contains [whatever]" but i can't seem to find that type of constraint.

**Comment by GeorgeMac** [comment]
Hey @csyding could the `IS ONE OF` operator work for your situation?

<img width="563" alt="Screenshot 2024-09-20 at 13 21 21" src="https://github.com/user-attachments/assets/64b0dbff-dc68-4a85-94a8-fa366a5ab73e">

Sorry, it is not documented yet (I don't think).
You can add multiple strings to compare against?

This might not work for your particular situation, where you want to send an array? (Instead of compare against one in a constaint).

One hacky option is that you could create a context with a key per group:
```
{
  "is_admin": "true",
  "is_accounting": "false",
  "is_underwriting": "true",
}
```

You could build that programatically in a loop and then write constraints like:

<img width="1066" alt="Screenshot 2024-09-20 at 13 28 52" src="https://github.com/user-attachments/assets/a682637b-f963-4cde-ab50-d9e0aa05e003">

Where you say `ANY` of the constraints match.
